### PR TITLE
Configure the Docker to run the project.

### PR DESCRIPTION
**Pull request by jhonathas**
Check results in: https://gist.github.com/jhonathas/7b0abd88591d1b3b01e8

### CODE REVIEW

**Comment by lucianosousa** [comment]
Can we use `.env.template` instead of `.env.docker` (I see duplicated code here).

Also, I suggest we configure the database user and password with env variables, then we can use the same database config file

**Comment by tauil** [comment]
I wouldn't put a Docker related file in the project. It's a particular choice of each developer. I would put this in a separated project. It's just my opinion. If the majority of maintainers agree with this, merge it.

**Comment by jhonathas** [comment]
I have more information on that template.

File: config/database.yml.docker [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  adapter: postgresql
+  database: call4paperz_development
+  username: postgres
+  password: 123456
+  host: postgres
+  min_messages: WARNING
+test:
+  adapter: postgresql
+  database: call4paperz_test
+  username: postgres
```
Comment (tauil) [inline config/database.yml.docker]: All of this should be ENV variables

**Comment by tauil** [comment]
Also database.yml.example or database.yml.template instead of .docker just like @lucianosousa said.

**Comment by jhonathas** [comment]
Tauil,

We have a Dockerfile at the root, you will be responsible for creating the image with the project Gems among other things.

The meaning is lost if we do not use that.

**Comment by jhonathas** [comment]
Separated in different files for already do this in the project, database.yml.travis

**Comment by jhonathas** [comment]
@lucianosousa @tauil 

O projeto atualmente já possui 2 templates de banco de dados. Apenas segui uma regra do que parecia convencionado. 

Além do mais, achei mais justo separar e deixar que o bootstrap criasse os arquivos certos do que ter que informar no readme o que um usuário deveria fazer. Acho que o Bootstrap tem que ser mastigado. Quando mais complicação na entrada menos gente consegue levantar e ajudar. 

Acho que o start da aplicação tem q ser simples. Assim como o start da forma comum, bin/setup, o novo modelo bin/docker_setup faz tudo. 

Segui o que já vinha sendo utilizado em relação ao duplicidade de arquivo e nos novos arquivos eu já deixei as configurações necessárias para que funcione direto, sem necessidade nenhuma de customização.

**Comment by tauil** [comment]
@jhonathas We've decided to move this Docker setup to a separate project. Heres the link: https://github.com/call4paperz/docker

Please, open this PR in that project.

Thanks!

**Comment by jhonathas** [comment]
blz

2015-10-21 20:37 GMT-02:00 Rafael Borgonovi Tauil <notifications@github.com>
:

> @jhonathas <https://github.com/jhonathas> We've decided to move this
> Docker setup to a separate project. Heres the link:
> https://github.com/call4paperz/docker
>
> Please, open this PR in that project.
>
> Thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/call4paperz/call4paperz/pull/118#issuecomment-150044005>
> .
>



-- 
Jhonathas Matos
--------------------------------------------------------
email: jhonathas@gmail.com

**Comment by lucianosousa** [comment]
changed for https://github.com/call4paperz/call4paperz_docker to better description

**Comment by tauil** [comment]
It was already inside call4paperz project. I don't think that the prefix `call4paperz_` is better description but nevermind. I wouldn't change it again.

**Comment by dukex** [comment]
A note about this PR(and the others about docker)

* The `docker-setup` file is not necessary, like rails-setup 
* The docker env needs to be configured in docker-compose.yml
* The Dockerfile needs to run rake db:create db:migrate
* I think is better use the rails image instead of ruby
* A common project directory is /usr/src/app
* I think the links needs to create a alias to work like: `postgres:postgres` that create a alias called postgres point to postgres container in /etc/hosts

**Comment by tauil** [comment]
Agreed. :+1:

**Pull request closed without merging**